### ISSUE TITLE
Provide the functionality to prevent sorting by a particular column

### ISSUE THREAD

**Issue by SergeyAlexeev** [DevExpress/devextreme-reactive#363]
For now, there is no way to cancel sorting by a particular column. As a workaround, a user can use the `headerCellTemplate` property of the [TableHeaderRow](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/table-header-row/) plugin.

```js
<TableHeaderRow
  headerCellTemplate={({ column }) => {
    if (column.name === 'age') {
      return <th>{column.title}</th>
    }
    return undefined;
  }}
  allowSorting
/>
```

[Demo](https://stackblitz.com/edit/react-ghdw79?file=index.js).

**Comment by vapurrmaid** [comment]
@SergeyAlexeev How might one go about this if using the Material UI. I'm assuming we have to change the <th> </th> to something else, but I'm stuck figuring out what.

**Comment by SergeyAlexeev** [comment]
You can use the [TableCell](https://material-ui-1dab0.firebaseapp.com/api/table-cell/) component.

**Comment by stclairdaniel** [comment]
I'm sorry, but I'm feeling a bit silly here. How can I use the TableCell component here to achieve the same thing as in the demo? I'm not sure what component to pass TableCell - the docs say "The component used for the root node. Either a string to use a DOM element or a component." but anything I try collapses the columns into one. Here's example code that doesn't work:

`this.headerCellTemplate = ({ column }) => {
      if (column.name === 'username') {
        return <TableCell component={'th'}>{column.title}</TableCell>;
      }
      return undefined;
    };`

![screen shot 2017-11-21 at 8 53 19 am](https://user-images.githubusercontent.com/14972752/33085447-8480b8f0-ce99-11e7-9f09-53c5b55fdfa3.png)

You can see the Username column disappears and the table header cell text collapses into the next column.

I must be missing something obvious about how to correctly pass the root component into TableCell. Any help is appreciated!

**Comment by SergeyAlexeev** [comment]
I can't reproduce the issue on my side. Would you please modify this [sample](https://stackblitz.com/edit/react-r9vxcp?file=index.js) to reproduce the problem?

**Comment by stclairdaniel** [comment]
[Here](https://stackblitz.com/edit/react-r9vxcp?embed=1&file=index.js) is a reproduction. 
Changes: 
Add the `allowSorting` prop to `<TableHeaderRow>`
Change `headerCellTemplate` to return a `<TableCell>` if the column is `name` - if we use `age`, the column disappears entirely.

**Comment by stclairdaniel** [comment]
@SergeyAlexeev Pinging for advice - almost done with our implementation here and this is the last little hiccup.

**Comment by SergeyAlexeev** [comment]
Thank you for the clarification! Now, I see the problem. As a workaround, I suggest you specify a custom cell template width:

```js
const headerCellTemplate = ({ column }) => {
  if (column.name === 'name') {
    return <TableCell component={'th'} style={{ paddingLeft: 24, width: 100 }}>{column.title}</TableCell>;
  }
  return undefined;
};
```

**Comment by SergeyAlexeev** [comment]
We researched the problem deeper and created a [pull request](https://github.com/DevExpress/devextreme-reactive/pull/536) that fixes it.
This fix will be available with the next update.

**Comment by stclairdaniel** [comment]
Hey @SergeyAlexeev - the fix looks good, but I must be missing something obvious. Looking at [#496](https://github.com/DevExpress/devextreme-reactive/issues/496) I see that render functions have been replaced with components, which is great. However, when using a Template like above, I could return undefined to use the default component, but now I can't do that anymore since it expects a component to be returned. How can I modify the above example using cellComponent instead of headerCellTemplate? I've been struggling with it for a while with no luck.

**Comment by SergeyAlexeev** [comment]
Hi,

Starting with v.1.0.0-beta.2, you can use default Grid components as follows:

```js
this.headerCellComponent = props => (
  return <TableHeaderRow.Cell {...props} />
);

...

<TableHeaderRow
  cellComponent={this.headerCellComponent}
/>
```

See [this](https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/grid-basics/) guide for more details.

**Comment by stclairdaniel** [comment]
@SergeyAlexeev Thanks so much, that worked like a charm! Everything looks great in the new update.

**Comment by SergeyAlexeev** [comment]
We've introduced such a functionality in [1.1.0-beta.1](https://github.com/DevExpress/devextreme-reactive/releases/tag/v1.1.0-beta.1). It's available through the `SortingState` plugin's `columnExtensions` property:
 
```js
<SortingState
  columnExtensions={[
    { columnName: 'name', sortingEnabled: false },
  ]}
/>
```
 
Moreover, other Grid features such as editing, filtering, grouping, and column visibility changing can be configured in a similar way. The state management plugins (`EditingState`, `FilteringState`, `GroupingState`, `TableColumnVisibility`) contain the related `columnExtensions` properties.
 
You can install React Grid v1.1.0-beta.1 with the `next` tag:

`npm i --save @devexpress/dx-react-core@next @devexpress/dx-react-grid@next`
`npm i --save @devexpress/dx-react-grid-bootstrap3@next`
`npm i --save @devexpress/dx-react-grid-material-ui@next`

Please check this out and send us your feedback if any!

**Comment by natenrb9** [comment]
I have updated to 1.1.0-beta.1, and I am trying to use sortingEnabled using the example here: 

https://github.com/DevExpress/devextreme-reactive/blob/master/packages/dx-react-demos/src/demo-sources/grid-sorting/disable-column-sorting.jsxt

And it is throwing the error:
React does not recognize the `sortingEnabled` prop on a DOM element.

Was this new prop fully implemented?

It appears to throw the same error for groupingEnabled and filteringEnabled...

**Comment by SergeyAlexeev** [comment]
The described functionality is available in React Grid starting with [v1.1.0](https://github.com/DevExpress/devextreme-reactive/releases/tag/v1.1.0) that is officially released.
